src/zenml/utils/package_utils.py

```diff
@@ -48,3 +50,26 @@ def is_latest_zenml_version() -> bool:
         return False
     else:
         return True
+
+
+def clean_requirements(requirements: List[str]) -> List[str]:
+    """Clean requirements list from redundant requirements.
+
+    Args:
+        requirements: List of requirements.
+
+    Returns:
+        Cleaned list of requirements
+    """
+    cleaned = {}
+    for req in requirements:
+        package = (
+            req.split(">=")[0]
```

Comment:
Are these pip requirements? In that case it could also be `~=` and `^=` I think?

Reply:
Yeah I think we need to handle these two use cases as well (the ~= (compatible release) and ^= (compatible release allowing pre-releases))

Reply:
Also I don't think we handle `package1>=1.0.0,<2.0.0` yet with this implementation. And there are a few other edge cases we're not handling